.circleci/config.yml

```diff
@@ -1,33 +1,26 @@
 version: 2.1
 
+# add parameters 
+
 orbs:
-  win: circleci/windows@2.4.0
+  win: circleci/windows@2.4.1
 
+# No windows executor is listed here since windows builds use win/default and modify
+# the Python version through the conda environment.
 executors:
-  circleci_machine_37:
-    docker:
-      - image: cimg/python:3.7-node
-    working_directory: ~/repo
-  circleci_machine_38:
-    docker:
-      - image: cimg/python:3.8-node
-    working_directory: ~/repo
-  circleci_machine_39:
-    docker:
-      - image: cimg/python:3.9-node
-    working_directory: ~/repo
-  circleci_machine_310:
+  docker:
+    parameters:
+      python_version:
+        type: string
     docker:
-      - image: cimg/python:3.10-node
+        - image: cimg/python:<<parameters.python_version>>-node
     working_directory: ~/repo
-  demo_executor:
-    docker:
-      - image: cimg/python:3.8
 
 commands:
   setup_python_env:
     description: Setup Python environment
     steps:
+      - checkout
       - run:
           name: Install Python
           command: python --version
```

Comment:
```suggestion
```

Reply:
This doesn't do anything except show the Python version, which is already given in a following step.

Reply:
Can't make a comment there, but looks like we can remove L31 also? Since `-r requirements.txt` is called from `test_requirements.txt` already.

Reply:
Would also be nice if we could remove the `cd ..` at the end 🙂